### DISCUSSION TITLE
Unintended artefacts in the generated sound?

### DISCUSSION THREAD

**Discussion by probonopd** [probonopd/MiniDexed#102] [Q&A]
Following the examples from the book "The Complete DX7" I noticed three things:

* Operating MiniDexed with the rotary encoder and display seems to be much more intuitive (at least to me) than using the original DX7. The book "The Complete DX7" has to use a lot of words to describe which buttons to press in which order and what to see on the display, whereas MiniDexed has a "normal" menu that is pretty much self-explanatory once you know what double-clicking, clicking, and rotating-while-keeping-button-pressed mean. Operation may be a bit slower than with the many keys of the original DX7, but I could imagine that many DX7 users never knew which buttons they had to press in the first place. Which makes our menu a winner! The shortcuts (rotating-while-keeping-button-pressed) are really valuable, I use them all the time. Hats off @rsta2!

* Unlike with the original DX7, even changes that affect the level of an operator are immediately audible on MiniDexed. Cool!

* Does MiniDexed produce unintended artefacts in the generated sound? I noticed this when doing exercise 25, which goes like this:
  * Disable all effects (by setting them to Off)
  * Disable all TGs except TG1 (by setting Channel to Off)
  * Set panning to center for TG1
  * Load an "Init voice" patch (e.g., from [ENO.syx](http://dxsysex.com/SYSEX_DX7/Guest/ENO.syx)
  * Set the Output Level of OP1-4 to 99 (leave the others on 0)
  * Set Freq Coarse of OP2 and OP4 to 2 (leave OP1 and OP3 at 1)
  * Set Osc Detune of OP3 and OP4 to +7 (leave OP1 and OP2 at 0)
  * Play C2

You should hear a beating sound. But on a Raspberry Pi 2 with HDMI I also hear something "interrupting" the sound briefly every ~1 second or so.

My hypothesis is that other stuff is running on the same CPU core which can lead to this effect.

TODO: Try to reproduce this with Dexed on a computer, and/or with MiniDexed on other Raspberry Pi/sound output combinations.

**Comment by rsta2** [comment]
Maybe the RPi 2 is "overloaded". Have you tried it with `fast=true`?

**Reply by probonopd** [reply]
Good idea. Will try. The effect is only noticeable under very specific circumstances, not in "real world" sounds. So doesn't warrant being seen as a "bug" imho. Just thought it would be worth sharing.

**Reply by probonopd** [reply]
> TODO: Try to reproduce this with Dexed on a computer

I do not hear it there. But nevermind, this is more a curiosity than anything.
